kratos/utilities/result_dabatase.cpp

```diff
@@ -0,0 +1,198 @@
+
+const EntityDatabase& VariableDatabase::GetEntityData(const IndexType EntityIndex) const
+{
+    KRATOS_DEBUG_ERROR_IF(EntityIndex == mData.size()) << "Incompatible size. EntityIndex: " << EntityIndex << ". Size: " << mData.size() << std::endl;
+    return mData[EntityIndex];
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+double VariableDatabase::GetValue(
+    const IndexType EntityIndex,
+    const double Time,
+    const SizeType ComponentIndex,
+    const SizeType GPIndex
+    ) const
+{
+    KRATOS_DEBUG_ERROR_IF(EntityIndex == mData.size()) << "Incompatible size. EntityIndex: " << EntityIndex << ". Size: " << mData.size() << std::endl;
+    return mData[EntityIndex].GetValue(Time, ComponentIndex, GPIndex);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void VariableDatabase::SetValues(
+    const Vector& rValuesX,
+    const Vector& rValuesY,
+    const IndexType EntityIndex,
+    const SizeType ComponentIndex,
+    const SizeType GPIndex
+    )
+{
+    auto& r_database = mData[EntityIndex];
+    r_database.SetValues(rValuesX, rValuesY, ComponentIndex, GPIndex);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void VariableDatabase::Clear()
+{
+    // Clear stored databases
+    for (auto& r_database : mData) {
+        r_database.Clear();
+    }
+
+    // Clear the container
+    mData.clear();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ResultDatabase::Initialize(
+    const std::vector<IndexType>& rVariablesIndexes,
+    const std::vector<IndexType>& rValuesSizes,
+    const SizeType NumberOfEntites,
+    const SizeType NumberOfGP
+    )
+{
+    // If no variables we skip
+    if (rVariablesIndexes.size() == 0) {
+        return void();
+    }
+
+    KRATOS_ERROR_IF_NOT(rVariablesIndexes.size() == rValuesSizes.size()) << "Inconsistent sizes in the values sizes and the variable indexes" << std::endl;
+
+    // Auxiliar lambda to generate vectors of tables
+    auto table_generator =[](const SizeType NumberOfEntites, const SizeType NumberOfComponents, const SizeType NumberOfGP){std::vector<Table<double, double>*> aux_1(NumberOfComponents, nullptr); EntityDatabase aux_2(NumberOfGP, aux_1); VariableDatabase data(NumberOfEntites, aux_2); for (IndexType k = 0; k < NumberOfEntites; ++k){for (IndexType j = 0; j < NumberOfGP; ++j){ for (IndexType i = 0; i < NumberOfComponents; ++i){ data[k][j][i] = new Table<double, double>();}}}; return data;};
```

Comment:
please refactor more readable
And again, why manual memory management with `new`?

Reply:
Still working on it. I am having problems with delete and for using unique_ptr

Reply:
![imagen](https://user-images.githubusercontent.com/19991680/83663847-7f3dfd80-a5c9-11ea-88dc-7353007e29df.png)
@philbucher I will try one more thing, otherwise we will need to live with shared_ptr. For some reason Table is not compatible with unique_ptr (and I tried many things), i even tried to move the copy constructor fron  unaccessible to regular one

Reply:
For me ok
But were you using a shared ptr so far?

Reply:
And now I see the delete is problematic